src/Validator.ts

```diff
@@ -134,14 +144,12 @@ export default (data: ArrayBuffer | ArrayBufferLike): ValidationResults => {
   validationResult.memoryUsage = validationResult.commandCount / MEMORY_LIMIT;
 
   if(validationResult.memoryUsage > 1) {
-    const memoryUsageFormatted = parseFloat((validationResult.memoryUsage * 100).toFixed(2));
-    const memError = `Used ${memoryUsageFormatted}% of available memory! Sequence uses ${validationResult.commandCount} commands, but the maximum allowed is ${MEMORY_LIMIT}!`;
-    if(validationResult.error) {
-      validationResult.error += ', ' + memError;
-    } else {
-      validationResult.error = memError
-    }
+    validationResult.errors!.push(ErrorType.Memory)
   }
 
+  if (validationResult.errors!.length == 0) {
+    validationResult.errors = undefined;
+  };
```

Comment:
No need for this either - the documentation can be updated to say `if errors is empty, file is valid`